### PR TITLE
Improve Network Integration Tests

### PR DESCRIPTION
**Pull request by mateoflorido**
## Overview
Make the network tests more generic. This will allow us to test on any implementation of the network feature.

## Changes
- Removed Cilium bits from the Network integration tests
- Drive-by fixes for the `x-wait-for` error messages and improvements to the StatusInterface docs.

### CODE REVIEW

File: tests/integration/tests/test_network.py [hunk 1]
```diff
@@ -78,20 +36,28 @@ def test_network(session_instance: harness.Instance):
...
+    p = session_instance.exec(
+        [
+            "curl",
+            "-s",
+            f"http://{podIP}",
+        ],
+        capture_output=True,
+    )
+
+    assert "Welcome to nginx!" in p.stdout.decode()
```
Comment (louiseschmidtgen) [inline tests/integration/tests/test_network.py:63]:
```suggestion
    assert "Welcome to nginx!" in p.stdout.decode(), "No NGINX welcome message found"
```

**Review by louiseschmidtgen — APPROVED** [review]
Little suggestion, LGTM

File: tests/integration/tests/test_network.py [hunk 2]
```diff
@@ -78,20 +36,28 @@ def test_network(session_instance: harness.Instance):
...
             "json",
         ],
         capture_output=True,
     )
-    assert "nginx" in p.stdout.decode().strip()
+
+    out = json.loads(p.stdout.decode())
+
+    assert len(out["items"]) > 0, "No NGINX pod found"
+    podIP = out["items"][0]["status"]["podIP"]
```
Comment (neoaggelos) [inline tests/integration/tests/test_network.py:52]: this needs some sort of retry in case the pod is not yet ready?

**Review by bschimke95 — COMMENTED** [review]
Great work Mateo. 
Two things:
* Please rebase on https://github.com/canonical/k8s-snap/pull/475 on ce this is merged, so that we can check if this works for all flavours.
* This covers `test_network`, we also need to update the gateway, ingress and loadbalancer tests. Should we add this changes here or just create follow-up cards?

File: tests/integration/tests/test_network.py (hunk 2 above)
Comment (mateoflorido) [reply tests/integration/tests/test_network.py:52]: Sure, lines 11-33 should wait until the pod reaches `condition=ready`. Do you want to retry the other parts of the test as well?

**Comment by louiseschmidtgen** [comment]
> Great work Mateo. Two things:
> 
>     * Please rebase on [Add moonray integration tests CI #475](https://github.com/canonical/k8s-snap/pull/475) on ce this is merged, so that we can check if this works for all flavours.
> 
>     * This covers `test_network`, we also need to update the gateway, ingress and loadbalancer tests. Should we add this changes here or just create follow-up cards?

The gateway, ingress and loadbalancer tests are assigned to Etienne and I 👍🏼

File: tests/integration/tests/test_network.py (hunk 2 above)
Comment (bschimke95) [reply tests/integration/tests/test_network.py:52]: The pod should always be ready at this point. As Mateo already said, we wait for the pod to come up further above.

**Review by bschimke95 — APPROVED** [review]
LGTM, great stuff @mateoflorido

**Pull request merged**